newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesGrid.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import * as React from 'react';
+import { type ExampleShortHeader } from '../../../../Utils/GDevelopServices/Example';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import ButtonBase from '@material-ui/core/ButtonBase';
+import GridList from '@material-ui/core/GridList';
+import GridListTile from '@material-ui/core/GridListTile';
+import { Column, Line } from '../../../../UI/Grid';
+import {
+  useResponsiveWindowWidth,
+  type WidthType,
+} from '../../../../UI/Reponsive/ResponsiveWindowMeasurer';
+import { CorsAwareImage } from '../../../../UI/CorsAwareImage';
+import Text from '../../../../UI/Text';
+import { sendExampleDetailsOpened } from '../../../../Utils/Analytics/EventSender';
+
+const getColumnsFromWidth = (width: WidthType) => {
+  switch (width) {
+    case 'small':
+      return 1;
+    case 'medium':
+      return 4;
+    case 'large':
+    default:
+      return 6;
+  }
+};
+const MAX_COLUMNS = getColumnsFromWidth('large');
+const MAX_TILE_SIZE = 300;
+
+const styles = {
+  examplesContainer: {
+    marginTop: 25,
+  },
+  grid: {
+    maxWidth: (MAX_TILE_SIZE + 2 * 8) * MAX_COLUMNS, // Avoid tiles taking too much space on large screens.
+  },
+  buttonStyle: {
+    textAlign: 'left',
+    width: '100%',
+  },
+  imageContainer: {
+    position: 'relative',
+    width: '100%',
+  },
+  thumbnailImageWithDescription: {
+    objectFit: 'contain',
+    verticalAlign: 'middle',
+    backgroundColor: 'black',
+    width: 'calc(100% - 2px)', // Not full because of border.
+    borderRadius: 8,
+    border: '1px solid lightgrey',
+    // Prevent cumulative layout shift by enforcing
+    // the 16:9 ratio.
+    aspectRatio: '16 / 9',
+  },
+};
+
+// Styles to give the impression of pressing an element.
+const useStylesForTile = makeStyles(theme =>
+  createStyles({
+    tile: {
+      borderRadius: 8,
+      padding: 4,
+      height: 'auto',
+      '&:focus': {
+        backgroundColor: theme.palette.action.hover,
+      },
+      '&:hover': {
+        backgroundColor: theme.palette.action.hover,
+      },
+    },
+  })
+);
+
+type ExamplesGridProps = {|
+  exampleShortHeaders: Array<ExampleShortHeader>,
+  limit?: number,
+  onOpen: (exampleShortHeader: ExampleShortHeader) => void,
+|};
+
+const ExamplesGrid = ({
+  exampleShortHeaders,
+  limit,
+  onOpen,
+}: ExamplesGridProps) => {
+  const windowWidth = useResponsiveWindowWidth();
+  const tileClasses = useStylesForTile();
+  const exampleShortHeadersToDisplay = limit
+    ? exampleShortHeaders.slice(0, limit)
+    : exampleShortHeaders;
+
+  return (
+    <div style={styles.examplesContainer}>
+      <Line noMargin>
+        <GridList
+          cols={getColumnsFromWidth(windowWidth)}
+          style={styles.grid}
+          cellHeight="auto"
+          spacing={16}
```

Comment:
Should this value be at the top of the file and linked to other values? (like the 8 I see in the grid max width)

Reply:
yup, that makes sense